### PR TITLE
new d3js cheat sheet

### PR DESCRIPTION
**Pull request by navneetzz**
What does your Instant Answer do?
My Instant answer provide details of the basic syntax used in D3js

What is the data source for your Instant Answer? (Provide a link if possible)
Data Driven Document (http://d3js.org/)

Why did you choose this data source?
Because it's official site for D3js

Are there any other alternative (better) data sources?
No

Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)
Data visualizations field programmers will get benefits out of it.

Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?
No, this not connected to DuckDuckHack Instant Answer.

Are you having any problems? Do you need our help with anything?
No I'm not having any problem.

Where did you hear about DuckDuckHack? (For first time contributors)
Social media sites.

What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)
screenshot 2016-01-08 22 51 15
https://duck.co/ia/view/d3_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/d3js.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
+{
+    "id": "d3js_cheat_sheet",
+    "name": "d3js Cheat Sheet",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/d3js.json]: lets remove "Cheat Sheet" from the name

File: share/goodie/cheat_sheets/json/d3js.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    "id": "d3js_cheat_sheet",
+    "name": "D3js",
+
+    "description": "Data Driven Document syntax",
+    "metadata": {
+        "sourceName": "DataDrivenDocument",
+        "sourceUrl": "http://d3js.org/"
+    },
+    "aliases": [
+        "d3", "datadrivendocument"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/d3js.json]: @navneetzz Lets give space betwen the words or just add another alias with spaces between the words

File: share/goodie/cheat_sheets/json/d3js.json [hunk 3]
```diff
@@ -0,0 +1,102 @@
+{
+    "id": "d3js_cheat_sheet",
+    "name": "D3js",
+
+    "description": "Data Driven Document syntax",
+    "metadata": {
+        "sourceName": "DataDrivenDocument",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/d3js.json]: @navneetzz  I would use "D3js" as source name

File: share/goodie/cheat_sheets/json/d3js.json (hunk 2 above)
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/d3js.json]: The website title is " Data-Driven Documents" but we could make that short by using "D3js"

Comment (mintsoft) [inline share/goodie/cheat_sheets/json/d3js.json]: I think very few people (unless they've actually investigated) know that `d3` is short for `data driven documents`

Comment (navneetzz) [inline share/goodie/cheat_sheets/json/d3js.json]: @minsoft what you suggest ?

File: share/goodie/cheat_sheets/json/d3js.json [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    "id": "d3js_cheat_sheet",
+    "name": "D3js",
+
+    "description": "Data Driven Document syntax",
+    "metadata": {
+        "sourceName": "D3js",
+        "sourceUrl": "http://d3js.org/"
+    },
+    "aliases": [
+        "d3", "datadrivendocument", "data-driven document", "data driven document"
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/d3js.json]: I'd probably add `d3 js` too

File: share/goodie/cheat_sheets/json/d3js.json [hunk 5]
```diff
@@ -0,0 +1,102 @@
+{
+    "id": "d3js_cheat_sheet",
+    "name": "D3js",
+
+    "description": "Data Driven Document syntax",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/d3js.json]: I'd probably change this to `Data Driven Document (d3js) syntax`

**Comment by gautamkrishnar** [comment]
@mintsoft :+1:

**Comment by navneetzz** [comment]
@mintsoft Travis is giving bug now also ?

**Comment by moollaza** [comment]
@navneetzz you have a JavaScript style comment in your JSON -- this is not valid and is causing a parsing error. Please remove that and then the test should pass again.

File: share/goodie/cheat_sheets/json/d3js.json [hunk 6]
```diff
@@ -0,0 +1,102 @@
+{
+    "id": "d3js_cheat_sheet",
+    "name": "D3js",
+
+    "description": "Data Driven Document (d3js) syntax", // changed syntax from Data Driven Document syntax to Data Driven Document (d3js) syntax
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/d3js.json]: Whoops, there are no "comments" in JSON -- please remove this.

Comment (navneetzz) [inline share/goodie/cheat_sheets/json/d3js.json]: @moollaza  removed the comment but now it's giving problem in deploying.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=d3js cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/d3_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@navneetzz this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Pull request merged**